platform/ios/src/MGLMapView.mm

```diff
@@ -4683,6 +4676,47 @@ - (void)updateAnnotationViews
         if (annotationView)
         {
             annotationView.center = [self convertCoordinate:annotationContext.annotation.coordinate toPointToView:self];
+            // Pin the callout view to the view annotation
+            UIView <MGLCalloutView> *calloutView = self.calloutViewForSelectedAnnotation;
+            if (calloutView && calloutView.representedObject == annotationContext.annotation) {
+                CGPoint point = annotationView.center;
+                point.x += annotationView.centerOffset.dx;
+                point.y += annotationView.centerOffset.dy;
+                point.y -= annotationView.bounds.size.height/2.0f;
```

Comment:
`MinY` from the frame is not what I'm trying to achieve here but `MidY` from the bounds would work.